lib/src/expression/micro.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:angular_ast/angular_ast.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:quiver/core.dart';
+
+const _listEquals = const ListEquality();
+
+/// Returns whether [expression] is a special Angular *-star expression.
+///
+/// This means it won't parse correctly with the standard expression parser, and
+/// [parseMicroExpression] is needed to de-sugar the expression into its
+/// multiple intents.
+bool isMicroExpression(String expression) => false;
+
+/// Returns a de-sugared [NgMicroExpression] from [expression] on [directive].
+///
+/// Given a string like `"let x of items; trackBy: foo; let i = index"`:
+///
+/// - Treats the first statement as "reference x, ngForOf=items".
+/// - Treats second as "ngForTrackBy=foo".
+/// - Treats last as "reference i = index".
+NgMicroExpression parseMicroExpression(
+  String expression,
+  String directive, {
+  bool deSugarPipes: true,
+  String sourceUrl: '/test.dart',
+}) {
+  final assignments = <ReferenceAst>[];
+  final properties = <PropertyAst>[];
+  final parts = expression.split(';');
+
+  // Parse the first statement, which is special.
+  if (!parts[0].startsWith('let')) {
+    throw new FormatException('Expected "let"', expression, 0);
+  }
+  var parse = parts[0].split(' ');
+  assignments.add(new ReferenceAst(parse[1]));
+  var name = parse[2];
+  name = directive + name[0].toUpperCase() + name.substring(1);
+  properties.add(
+    new PropertyAst(
+      name,
+      new ExpressionAst.parse(
+        parse[3],
```

Comment:
could contain spaces, ie `let x of myMethod(arg1, arg2)`

Also, see work here that @mk13 has done related to trailing expressions, which you'll need to check: https://github.com/dart-lang/angular_analyzer_plugin/pull/94/files#diff-dd70e04facf5d905f5ff348ad54053f3R987